### ISSUE TITLE
Different variants after regenotyping

### ISSUE THREAD

**Issue by meiers** [freebayes/freebayes#190]
Hi there,
I am wondering about this behaviour of freebayes when re-genotyping (-@).
I call variants on two samples separately and get the following vcfs:
sample "VRG"

```
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  unknown
chr2L   5038    .       TCGACAATGCACGACAGAGGAAGCAGAACAG TTAA    1048.99 .       1/1:41:1:37:38:1269:-106.959,-12.1824,0
chr2L   5073    .       TTAGATTGCCTCTCA TA      825.754 .       1/1:42:2:68:39:968:-84.8235,-11.863,0
...
```

sample "CROSS"

```
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  unknown
chr2L   5038    .       TCGACAATGCACGACAGAGGAAGCAGAACAG TTAA    1532.93 .       1/1:58:1:35:56:1887:-158.061,-20.2744,0
chr2L   5073    .       TTAGATTGCCTCTCA TA      1258.87 .       1/1:60:0:0:56:1476:-129.023,-17.4597,0
...
```

Looks like these variants are homozygous in both samples (kind of what I expected). Then I merge the two VCFs, strip off the sample columns and use this input for regenotyping both samples. This is how the merge looks:

```
#CHROM  POS     ID      REF     ALT     QUAL    FILTER
chr2L   5038    .       TCGACAATGCACGACAGAGGAAGCAGAACAG TTAA    1532.93 .
chr2L   5073    .       TTAGATTGCCTCTC  T       1258.87 .
...
```

The second variant has been altered, but that's not the point. Next I use `freebayes -@ -l` on both samples (separately) to get a genotype on the merged calls. This is how it looks afterwards:
sample "VRG"

```
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  virginizer
chr2L   5038    .       TCGACAAT        TT      0.952231        .       0/0:3:2:61:0:0:0,-2.10721,-2.41702
chr2L   5065    .       ACAGA   AA      3.37512 .       0/0:3:3:90:0:0:0,-0.90309,-1.68535
chr2L   5073    .       TTAGATTGCCTCTCA TA      825.754 .       1/1:42:2:68:39:968:-84.8235,-11.863,0
...
```

Sample "CROSS" looks similar. 
Long story short, what happened to the original variant at pos 5038? I thought that freebayes would just take the input variants from `-@` and determine alt/ref support for them.
Thanks for taking a look at this!
Best,
Sascha
